doc/adr/0002-use-meta-yaml-to-track-feedstock-metadata.md

```diff
@@ -0,0 +1,158 @@
+# 1. Use meta.yaml to track feedstock metadata
+
+Date: 2021-03-27
+
+## Status
+
+Proposed
+
+## Context
+
+We need a way to keep track of metadata associated with each feedstock repo.
+
+### Goals
+
+- The metadata should describe everything we need to know about the recipes that is not contained in the recipe python code itself.
+- It should facilitate generation of a STAC collection for the data that comes out of the recipe.
+- It should tell the bakery everything it needs to know to run the recipe.
+
+### Inspiration
+
+- Conda forge [meta.yaml](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html).
+- [STAC collection example](https://github.com/radiantearth/stac-spec/blob/master/examples/collection.json)
+
+
+## Decision
+
+We will track metadata in a `meta.yaml` which lives at the top level of a feedstock repo.
+The format and contents of this file are specified as follows.
+
+### Top Level Data
+
+```yaml
+id: noaa-oisst  # top-level ID for the feedstock. must match feedstock repo name
+title: "NOAA Optimum Interpolated SST"
+description: "Analysis-ready Zarr datasets derived from NOAA OISST NetCDF"
+pangeo_forge_recipe_version: "0.1"
+pangeo_forge_metadata_spec_version: "1"
+```
+
+### `recipes` section
+
+
+The `recipes` section explains what recipes are contained in the repo. For the simple case of one recipe, it looks like this
+
+```yaml
+recipes:
+  - id: avhrr-only
+    module: recipe.py  # the module where to find the recipe
+    name: recipe  # the name of the object to import
+```
+
+To find the code for this recipe, the parsing library would have to do something link
+
+```python
+from recipe import recipe
+```
+If the repo provides multiple recipes in the same module, it might look like this
+
+```yaml
+recipes:
+  - id: avhrr-only
+    module: recipe.py
```

Comment:
I suggest here (and below) to refer not to a filename, but an importable name. Since the current directory is included in python's path, this would amount to recipe.py -> recipe. This allows referencing recipe code within packages.

In addition, module and name could be combined in the style of entrypoints: "package.module:name". Having multiple import levels is optional. In fact, in full the entrypoints spec goes like "id = package.module:name", for reference.

We should be clear here what requirements the object represented by name has.

Reply:
I have been unable to find good documentation on the entrypoints syntax. The best I found was this: https://packaging.python.org/guides/distributing-packages-using-setuptools/#entry-points

Is there something more comprehensive out there?